script/upgrades/MU01/governance/MU01-CGP-Phase2.sol

```diff
@@ -0,0 +1,256 @@
+   */
+  function setAddresses() public {
+    cUSD = contracts.celoRegistry("StableToken");
+    cEUR = contracts.celoRegistry("StableTokenEUR");
+    cBRL = contracts.celoRegistry("StableTokenBRL");
+    celo = contracts.celoRegistry("GoldToken");
+    bridgedUSDC = contracts.dependency("BridgedUSDC");
+  }
+
+  /**
+   * @dev Setups up various configuration structs.
+   *      This function is called by the governance script runner.
+   */
+  function setUpConfigs() public {
+    // Create pool configurations
+    cUSDCeloConfig = Config.cUSDCeloConfig(contracts, 2);
+    cEURCeloConfig = Config.cEURCeloConfig(contracts, 2);
+    cBRLCeloConfig = Config.cBRLCeloConfig(contracts, 2);
+    cUSDUSDCConfig = Config.cUSDUSDCConfig(contracts, 2);
+
+    // Push them to the array
+    poolConfigs.push(cUSDCeloConfig);
+    poolConfigs.push(cEURCeloConfig);
+    poolConfigs.push(cBRLCeloConfig);
+    poolConfigs.push(cUSDUSDCConfig);
+
+    // Set the exchange ID for the reference rate feed
+    for (uint i = 0; i < poolConfigs.length; i++) {
+      referenceRateFeedIDToExchangeId[poolConfigs[i].referenceRateFeedID] = getExchangeId(
+        poolConfigs[i].asset0,
+        poolConfigs[i].asset1,
+        poolConfigs[i].isConstantSum
+      );
+    }
+  }
+
+  function run() public {
+    prepare();
+    address governance = contracts.celoRegistry("Governance");
+    ICeloGovernance.Transaction[] memory _transactions = buildProposal();
+
+    vm.startBroadcast(Chain.deployerPrivateKey());
+    {
+      createProposal(_transactions, "MU01-Phase2", governance);
+    }
+    vm.stopBroadcast();
+  }
+
+  function buildProposal() public returns (ICeloGovernance.Transaction[] memory) {
+    require(transactions.length == 0, "buildProposal() should only be called once");
+
+    proposal_createExchanges();
+    proposal_configureTradingLimits();
+    proposal_configureV1Exchanges();
+
+    return transactions;
+  }
+
+  /**
+   * @notice This function generates the transactions required to create the
+   *         BiPoolManager exchanges (cUSD/CELO, cEUR/CELO, cBRL/CELO, cUSD/bridgedUSDC)
+   */
+  function proposal_createExchanges() private {
+    address payable biPoolManagerProxy = contracts.deployed("BiPoolManagerProxy");
+    bool biPoolManagerInitialized = BiPoolManagerProxy(biPoolManagerProxy)._getImplementation() != address(0);
+    if (biPoolManagerInitialized) {
+      bytes32[] memory existingExchangeIds = IBiPoolManager(contracts.deployed("BiPoolManagerProxy")).getExchangeIds();
+      if (existingExchangeIds.length > 0) {
+        console.log("Destroying existing exchanges: ", existingExchangeIds.length);
+        for (uint256 i = existingExchangeIds.length; i > 0; i--) {
+          transactions.push(
+            ICeloGovernance.Transaction(
+              0,
+              contracts.deployed("BiPoolManagerProxy"),
+              abi.encodeWithSelector(IBiPoolManager(0).destroyExchange.selector, existingExchangeIds[i - 1], i - 1)
```

Comment:
nit: I think looping from `existingExchangeIds.length - 1` to `0` (inclusive) is a bit more readable than having to do `i - 1` for the index

Reply:
This logic was taken from the phase1 script, so I'd keep it consistent